pkg/util/autopilot.go

```diff
@@ -55,13 +59,20 @@ func IsAutopilotManagedNamespace(o client.Object) bool {
 }
 
 // IsGKEAutopilotCluster returns if the cluster is an autopilot cluster.
-// It leverages the existence of the workload-defaulter mutating webhook configuration, which exists and only exists on the Autopilot clusters.
+// It determines an Autopilot cluster by the presence of autopilot webhooks.
 func IsGKEAutopilotCluster(c client.Client) (bool, error) {
-	err := c.Get(context.Background(), client.ObjectKey{Name: autopilotWebhook}, &admissionregistrationv1.MutatingWebhookConfiguration{})
-	if err == nil {
-		return true, nil
+	var autopilotWebhookNotFoundCount int
+	var err error
+	for _, autopilotWebhook := range autopilotWebhooks {
+		err = c.Get(context.Background(), client.ObjectKey{Name: autopilotWebhook}, &admissionregistrationv1.MutatingWebhookConfiguration{})
+		if err == nil {
+			return true, nil
+		}
+		if apierrors.IsNotFound(err) {
+			autopilotWebhookNotFoundCount++
```

Comment:
nit: you don't need the `autopilotWebhookNotFoundCount` variable. You can `return false, err` `if !apierrors.IsNotFound(err)`, and `return false, nil` outside the for-loop.

Reply:
If the first iteration encounters an error other than `NotFoundErr`, we should still go to the second iteration.

Reply:
If the first iteration failed with an error other than `NotFoundErr`, that seems to indicate some k8s-apiserver error. Should we return the error and requeue the reconciliation request?

With the current approach, if the Get call for both webhooks failed, the first error will be overridden by the second one.

Reply:
If the first iteration failed with an error other than NotFoundErr, it it still possible that the second iteration succeeds and identify the cluster as an Autopilot cluster. In this case, I don't think it is necessary to return the error in the first iteration.

If both Get calls failed with errors other than NotFoundErr, it is very likely that they have the same root cause. Therefore, it should be okay to just return the error in the second iteration.

Reply:
SG.